tests/src/test/java/org/teavm/classlib/java/util/TreeSetTest.java

```diff
@@ -0,0 +1,178 @@
+        try {
+            assertFalse(it.hasNext());
+            it.next();
+            fail();
+        } catch (NoSuchElementException e) {
+            // ok
+        }
+        subset = reversed.tailSet("2");
+        it = subset.iterator();
+        assertTrue(it.hasNext());
+        assertEquals("2", it.next());
+        assertTrue(it.hasNext());
+        assertEquals("1", it.next());
+        try {
+            assertFalse(it.hasNext());
+            it.next();
+            fail();
+        } catch (NoSuchElementException e) {
+            // ok
+        }
+    }
+
+    @Test
+    public void testSequencedCollectionIterator() {
+        SortedSet<String> set = new TreeSet<>(List.of("a", "b", "c", "d")).reversed();
+        Iterator<String> it = set.iterator();
+        assertEquals("d", it.next());
+        assertEquals("c", it.next());
+        it.remove();
+        assertArrayEquals(new String[] { "d", "b", "a" }, set.toArray());
+        set = new TreeSet<>(List.of("a", "b", "c", "d")).reversed();
+        it = set.iterator();
+        assertEquals("d", it.next());
+        assertEquals("c", it.next());
+        it.remove();
+        assertEquals("b", it.next());
+        it.remove();
+        assertEquals("a", it.next());
+        try {
+            it.next();
+            fail();
+        } catch (NoSuchElementException e) {
+            // ok
+        }
+        assertArrayEquals(new String[] { "d", "a" }, set.toArray());
+    }
+
+    @Test
+    public void sequenceCollectionMethodsOnEmpty() {
+        var empty = new TreeSet<>();
+
+        try {
+            empty.getFirst();
+            fail();
+        } catch (NoSuchElementException e) {
+            // ok
+        }
+        try {
+            empty.getLast();
+            fail();
+        } catch (NoSuchElementException e) {
+            // ok
+        }
+
+        try {
+            empty.removeFirst();
+            fail();
+        } catch (NoSuchElementException e) {
+            // ok
+        }
+        try {
+            empty.removeLast();
+            fail();
+        } catch (NoSuchElementException e) {
+            // ok
+        }
+        try {
+            empty.addFirst("r");
+            fail();
+        } catch (UnsupportedOperationException e) {
+            // ok
+        }
+        try {
+            empty.addLast("r");
+            fail();
+        } catch (UnsupportedOperationException e) {
+            // ok
+        }
+    }
+
+    @Test
+    public void testMutationsIterator() {
+        SortedSet<String> set = new TreeSet<>(List.of("a", "b", "c", "d"));
+        set.add("e");
+        set.add("f");
+        Iterator<String> it = set.iterator();
+        assertEquals("a", it.next());
+        it.remove();
+        assertEquals("b", it.next());
+        it.remove();
+        assertEquals("c", it.next());
+        it.remove();
+        assertEquals("d", it.next());
+        it.remove();
+//      TODO fixme  assertEquals("e", it.next());
```

Comment:
Just pushed the fix for this issue, you can try to merge and uncomment this

Reply:
Works, thanks.